### PR TITLE
DSU-1250 Anonymize capitalized names in subject

### PR DESCRIPTION
**Pull request by obukhov-sergey**
See https://mailgun.atlassian.net/browse/DSU-1250
For provided examples the email was sent to address that has no names in it:

Screenshot from 2022-10-07 15-12-08

We can go over a list of most popular names but no list have all the names and no list has all last names.
It's error prone. For example Rome is both a name and a city. It's gonna affect current stats for a month: we keep stats and search it by anonymized subject, if the anonymization algorithm changes they won't match.

Nevertheless. The PR uses NLP library to find entities labeled as "PERSON" and replaces them with xxx.

### CODE REVIEW

File: anonymize/anonymize.go [hunk 1]
```diff
@@ -5,6 +5,9 @@ import (
 	"regexp"
 	"sort"
 	"strings"
+
+	"github.com/jdkato/prose/v2"
+	"github.com/mailgun/holster/errors"
```
Comment (vtopc) [inline anonymize/anonymize.go:10]:
```suggestion
	"github.com/mailgun/holster/v4/errors"
```

File: anonymize/anonymize.go [hunk 2]
```diff
@@ -13,16 +16,34 @@ var adjacentSecrets = regexp.MustCompile(`xxx(\sxxx)+`)
...
-		return src, nil
+		return s, nil
 	}
 	secret, err := or(tokens)
 	if err != nil {
-		return src, err
+		return s, err
+	}
+	s = secret.ReplaceAllString(s, "xxx")
+	s = adjacentSecrets.ReplaceAllString(s, "xxx")
```
Comment (vtopc) [inline anonymize/anonymize.go:32]: This `xxx` is used 3 times in the code, please, move it to some constant.

Comment (obukhov-sergey) [reply anonymize/anonymize.go]: good point, ty!

Comment (obukhov-sergey) [reply anonymize/anonymize.go]: done

**Review by vtopc — APPROVED** [review]
LGTM

**Pull request merged**